lib/Reader/abisignature.cpp

```diff
@@ -247,6 +247,20 @@ static Value *getFieldAddress(IRBuilder<> &Builder, Value *Base,
   return Address;
 }
 
+bool ABICallSignature::hasIndirectResultOrArg() const {
```

Comment:
I believe there's some paranoia in lowering that should detect this case and avoid the optimization (it looks generally for by-val memory arguments from the caller's frame).

Does this case only happen with SysV? I wanted to test there but still have trouble doing any significant testing on Linux.